### PR TITLE
Helper: Filter room with distance > limit

### PR DESCRIPTION
**Pull request by TooAngel**

### CODE REVIEW

**Comment by TooAngel** [comment]
DCBOT: 
This repository is under [democratic collaboration](https://github.com/TooAngel/democratic-collaboration) and will be merged automatically.

The merge decision is based on the outcome of the reviews:
 - `Approve` add the reviewer value (number of commits) to the `metric`
 - `Request changes` substract the reviewer value from the `metric`

 The merge will happen after `metric * 10 days` calculated from the last code change.

Please review the PR to make a good democratic decision.

Summoning some reviewers:
 - @Nevrdid: I just want to get this merged ASAP
 - @WhiteHalmos: Some warm words would be nice
 - @Somotaw: Please look at the code

**Comment by TooAngel** [comment]
DCBOT: A new review, yeah.

    Votes: 153/289
    Coefficient: 0.529411764706
    Hours to merge: 16
    Age in hours: 0

**Comment by TooAngel** [comment]
DCBOT: A new review, yeah.

    Votes: 189/289
    Coefficient: 0.653979238754
    Hours to merge: 9
    Age in hours: 1

**Comment by TooAngel** [comment]
DCBOT: A new review, yeah.

    Votes: 213/289
    Coefficient: 0.737024221453
    Hours to merge: 12
    Age in hours: 2

File: src/prototype_room_utils.js [hunk 1]
```diff
@@ -10,9 +10,14 @@ Room.prototype.sortMyRoomsByLinearDistance = function(target) {
...
   let roomName = this.name;
+  let filterByLinearDistance = function(object) {
+    let dist = Game.map.getRoomLinearDistance(roomName, object);
+    return dist <= limit;
+  };
+  roomsNames = _.filter(roomsNames, filterByLinearDistance);
   let sortByLinearDistance = function(object) {
     let dist = Game.map.getRoomLinearDistance(roomName, object);
-    return dist <= limit ? dist : 1000;
+    return dist;
```
Comment (Nevrdid) [inline src/prototype_room_utils.js]: it loop 2 times. I supose we can loop only once like that for exemple : 
```
   let sortByLinearDistance = function(object) {
     let dist = Game.map.getRoomLinearDistance(roomName, object);
    return dist <= limit ? dist : false;
  }
  let min = _.min(roomsNames, sortByLinearDistance)
  return min === Infinity ? undefined : min;
```

**Comment by TooAngel** [comment]
DCBOT: A new review, yeah.

    Votes: 264/291
    Coefficient: 0.907216494845
    Merging in -1 days 8 hours
    Age 1 days 13 hours

**Pull request merged**